cookbook/security/access_control.rst

```diff
@@ -54,12 +54,10 @@ Take the following ``access_control`` entries as an example:
 
             <config>
                 <!-- ... -->
-                <access-control>
-                    <rule path="^/admin" role="ROLE_USER_IP" ip="127.0.0.1" />
-                    <rule path="^/admin" role="ROLE_USER_HOST" host="symfony\.com$" />
-                    <rule path="^/admin" role="ROLE_USER_METHOD" method="POST, PUT" />
-                    <rule path="^/admin" role="ROLE_USER" />
-                </access-control>
+                <rule path="^/admin" roles="ROLE_USER_IP" ip="127.0.0.1" />
+                <rule path="^/admin" roles="ROLE_USER_HOST" host="symfony\.com$" />
+                <rule path="^/admin" roles="ROLE_USER_METHOD" methods="POST, PUT" />
+                <rule path="^/admin" roles="ROLE_USER" />
```

Comment:
`rule` and `access_control` do have the same semantic meaning in XML (actually `access_control` is the pluralised form of `rule`). In fact, nesting `access-control` and `rule` was a mistake we made in the documentation and only worked if we had only one rule.

The resulting array after processing the configuration would have been an array with `rule` as the key. This only worked because the extension just loops the array and doesn't deal with its keys. Though obviously this will break as soon as you have more than one rule.